### PR TITLE
Fix catkin_make failure (due to yaml-cpp deps) for mac os

### PR DESCRIPTION
**Pull request by njzhangyifei**
Fixed catkin_make failure (due to yaml-cpp deps) for mac os

Tested on Mac OS X yosemite w/ homebrew
Also working for Ubuntu 14.04.

However, yaml-cpp version is not checked for mac.

### CODE REVIEW

File: camera_calibration_parsers/CMakeLists.txt [hunk 1]
```diff
@@ -12,12 +12,20 @@ catkin_package(
...
+  SET(YAML_CPP_INCLUDE_DIR ${YAML_CPP_H_INCLUDE_DIR}/yaml-cpp)
+else()
+  pkg_check_modules(YAML_CPP yaml-cpp)
+  if(${YAML_CPP_VERSION} VERSION_GREATER 0.5)
+    add_definitions(-DHAVE_NEW_YAMLCPP)
+  endif()
+  link_directories(${YAML_CPP_LIBRARY_DIRS})
 endif()
 include_directories(${YAML_CPP_INCLUDE_DIRS})
-link_directories(${YAML_CPP_LIBRARY_DIRS})
```
Comment (njzhangyifei) [inline camera_calibration_parsers/CMakeLists.txt:20]: I think this is redundant, but I've moved this up to Line 26

**Comment by vrabaud** [comment]
Thx !

**Comment by wjwwood** [comment]
This is just not correct, the same logic should work for OS X as Linux. AFAIK there is nothing special about `yaml-cpp` on OS X.

@vrabaud I would revert this and ask for more information about the failure. I believe this treats a symptom rather than solving an issue.

**Comment by vrabaud** [comment]
@njzhangyifei , any opinion on that ?

**Comment by wjwwood** [comment]
I can do `pkg-config` on OS X 10.10:

```
% pkg-config --libs yaml-cpp
-L/usr/local/Cellar/yaml-cpp-0.3/0.3.0/lib -lyaml-cpp
```

I have `yaml-cpp-0.3` installed from Homebrew:

```
% brew list | grep yaml-cpp
yaml-cpp-0.3
```

ROS does not use `yaml-cpp` version `0.5` because of incompatibilities, maybe that is part of the confusion. I'd look into this further, but I really don't have time right now.

**Comment by njzhangyifei** [comment]
Adding a cmake file for yaml-cpp does resolve this issue for all platforms. My workaround is only a temporary one. Thank you for resolving this issue.

**Pull request merged**